### ISSUE TITLE
gtalksms permanently displays "Waiting for network to become available" toast, connected or not

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#46]
```
What steps will reproduce the problem?
1. Set tasker up with context - WiFi Connected, enter task as action intent to 
connect Tasker and exit task to disconnect, as per the wiki on public actions
2. Connect to WiFi network
3. Wait any amount of time [possibly sending messages]

What is the expected output? What do you see instead?
I'd expect to see this very temporarily if it drops out of signal, but it stays 
until I force close GTalk from the applications menu.

What version of the product are you using? On what operating system?
1.7, Oxygen ROM 2.0RC6 (based on AOSP 2.3)

Please provide any additional information below.
It's a bit of a pain, because it obstructs the on screen keyboard and any 
widgets on my home screen, and generally anything I try to use on my phone. 
Can't use gtalksms until it's fixed - really big shame.
```

Original issue reported on code.google.com by `philly.s...@gmail.com` on 12 Jan 2011 at 3:02

Attachments:
- [gtalksmslog.txt](https://storage.googleapis.com/google-code-attachments/gtalksms/issue-46/comment-0/gtalksmslog.txt)

**Comment by GoogleCodeExporter** [comment]
```
Update:It happens when disabling the wifi connection. Ignore step 3. I'm 
assuming that the wifi drops out, so GTalkSMS displays the toast then Tasker 
closes the connection too quickly afterwards and so the Toast is left open 
indefinitely.

I've temporarily fixed it by adding a 5 second wait before the disconnect 
action, although this is obviously less than ideal.
```

Original comment by `philly.s...@gmail.com` on 12 Jan 2011 at 3:16

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 12 Jan 2011 at 9:26
- Changed state: **Investigating**

**Comment by GoogleCodeExporter** [comment]
```
Can you please elaborate what "and exit task to disconnect," means?  How 
exactly is tasker configured to act on disconnection?  The correct thing would 
be to send a CONNECT intent with the disconnect extra set.  If tasker was 
managing to just kill the GTalkSMS process I could imagine there being a 
problem...
```

Original comment by `skippy.hammond@gmail.com` on 12 Jan 2011 at 10:58

**Comment by GoogleCodeExporter** [comment]
```
Could you please check to see if this is still an issue with the most recent 
version?  I've also updated the wiki to note that tasker-like apps should not 
terminate the application, so if we don't hear back soon will close this issue.

Thanks!

Mark
```

Original comment by `skippy.hammond@gmail.com` on 3 Feb 2011 at 7:10

**Comment by GoogleCodeExporter** [comment]
```
I've got a similar issue with a connection error popup.
Other toast messages show over of this one but I've killed the application to 
remove the stuck notification.

The app fails to connect the first time and was killed right away when the 
notification shows up... It's maybe an android toast management issue...
```

Original comment by `Florent....@gmail.com` on 3 Feb 2011 at 6:25

**Comment by GoogleCodeExporter** [comment]
```
Same problem on FroydVillain 1.7.3 on HTC Hero GSM.
Can you please provide a way to get rid of the message, until it's fixed ?
Thanks
```

Original comment by `sebastien.wains` on 14 Feb 2011 at 11:23

**Comment by GoogleCodeExporter** [comment]
```
Goto Preferences > Application settings > Status notification icon.

But you will also loose the icon status
```

Original comment by `Florent....@gmail.com` on 15 Feb 2011 at 9:58

**Comment by GoogleCodeExporter** [comment]
```
In fact, I'm wrong... That doesn't disable toast message.
You will have to wait for the next release.
```

Original comment by `Florent....@gmail.com` on 15 Feb 2011 at 1:15

**Comment by GoogleCodeExporter** [comment]
```
 I meant, how do one kill the toast message when it's already there ? It stayed overnight, had to reboot my phone this morning..
```

Original comment by `sebastien.wains` on 15 Feb 2011 at 7:53

**Comment by GoogleCodeExporter** [comment]
```
You have to kill the GTalkSMS process thanks to a process killer
```

Original comment by `Florent....@gmail.com` on 15 Feb 2011 at 8:12

**Comment by GoogleCodeExporter** [comment]
```
Tried that, didn't work :-(
```

Original comment by `sebastien.wains` on 15 Feb 2011 at 8:14

**Comment by GoogleCodeExporter** [comment]
```
Do you have stop GTalkSMS before (xmpp connection)? If it's started, kill not 
work.
```

Original comment by `Florent....@gmail.com` on 15 Feb 2011 at 8:21

**Comment by GoogleCodeExporter** [comment]
```
Not sure. I'll try that if it occurs again.
```

Original comment by `sebastien.wains` on 15 Feb 2011 at 8:22

**Comment by GoogleCodeExporter** [comment]
```
Issue 110 has been merged into this issue.
```

Original comment by `Florent....@gmail.com` on 23 Feb 2011 at 2:44

**Comment by GoogleCodeExporter** [comment]
```
I have been able to reproduce this issue every time my pass expired on WLAN 
network with frontpage. I need to kill the process in order to get rid of the 
toast.
```

Original comment by `piot...@gmail.com` on 28 Feb 2011 at 11:16

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 28 Feb 2011 at 11:28

**Comment by GoogleCodeExporter** [comment]
```
Issue 112 has been merged into this issue.
```

Original comment by `Florent....@gmail.com` on 28 Feb 2011 at 11:30

**Comment by GoogleCodeExporter** [comment]
```
It's not really a bug, it's more a problem of UX.
It just happen to lose connection (subway, wifi-3G-wifi_3G, tunnels, ...), thus 
displaying a Toast EVERY time you loose it and reconnect is just a bad UX 
pattern.
Imagine if all application do that...

it's a pity because the app itslef works great and is very usefull.
```

Original comment by `gregory....@gmail.com` on 28 Feb 2011 at 11:38

**Comment by GoogleCodeExporter** [comment]
```
We will keep toast messages only for xmpp protocol errors (login, password, 
server...)
```

Original comment by `Florent....@gmail.com` on 28 Feb 2011 at 8:18
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 28 Feb 2011 at 8:47
- Changed state: **WaitingForRelease**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v1.9.2
```

Original comment by `Florent....@gmail.com` on 28 Feb 2011 at 9:43
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 7:51
- Changed state: **Deployed**
